lib/widgets/app_bar.dart

```diff
@@ -6,14 +6,44 @@ import 'store.dart';
 ///
 /// This should be used for most of the pages with access to [PerAccountStore].
 class ZulipAppBar extends AppBar {
+  /// Creates our Zulip custom app bar based on [AppBar].
+  ///
+  /// [buildTitle] is passed a `willCenterTitle` callback that can answer
+  /// whether the underlying [AppBar] will decide to center [title]
+  /// based on [centerTitle], the theme, the platform, and [actions].
+  /// Useful if [title] is a container whose children should align the same way,
+  /// such as a [Column] with multiple lines of text.
+  // TODO send a PR upstream to replace our `willCenterTitle` code
   ZulipAppBar({
     super.key,
     super.titleSpacing,
-    required super.title,
+    required Widget? Function(bool Function(ThemeData) willCenterTitle) buildTitle,
```

Comment:
It took me a while to parse how this works. Maybe we can try to implement this as a widget? A concept would be like:

```dart
class _TitleBuilder extends StatelessWidget {
  const _TitleBuilder({
    required this.centerTitle,
    required this.actionsCount,
    required this.buildTitle,
  });

  final bool? centerTitle;
  final int actionsCount;
  final Widget Function(bool centerTitle) buildTitle;

  // Copied from [AppBar._getEffectiveCenterTitle].
  bool willCenterTitle(BuildContext context) {
     final theme = Theme.of(context);
     bool platformCenter() {
       switch (theme.platform) {
         case TargetPlatform.android:
         case TargetPlatform.fuchsia:
         case TargetPlatform.linux:
         case TargetPlatform.windows:
           return false;
         case TargetPlatform.iOS:
         case TargetPlatform.macOS:
           return actionsCount == 0 || actionsCount < 2;
       }
     }

     return centerTitle
       ?? theme.appBarTheme.centerTitle
       ?? platformCenter();
  }

  @override
  Widget build(BuildContext context) {
    return buildTitle(willCenterTitle(context));
  }
}
```

Letting `buildTitle` take a `bool` simplifies the API a bit and with a widget we can replace the callback constructed below with a method.

Reply:
Oh nice, this is great! Thanks for the idea.